### ISSUE TITLE
Add support for "additionalProperties" in multipart/form-data request bodies

### ISSUE THREAD

**Issue by adam-homeboost** [Rhosys/openapi-explorer#176]
Openapi allows an object to declare that it will accept non-specified "extra" properties by declaring "additionalProperties" to be true.  This is not a problem when executing json payloads from the explorer, but for GET requests and multipart/form-data requests, the form constructed by the explorer is not usable to set these fields.

Here is an image upload example, using multipart/form-data POST.

<img width="624" alt="image" src="https://github.com/Rhosys/openapi-explorer/assets/127809317/338830b2-ae3e-425c-9953-970ceb2edb6f">

Note that the ability to add extra fields is clearly recognized and displayed (the any-key element), however there is no ability to set the actual key value and there is no ability to set more than one extra parameter.

**Comment by wparad** [comment]
GET requests don't have bodies, so I'm not totally sure what you would be expecting there. From your screen shot I can see that there must be a bug for multipart/form-data bodies. Can you provide a minimal repo spec so that we can triage and begin work on fixing this?

Thanks,
Warren

**Comment by adam-homeboost** [comment]
GET requests can have query string parameters and the openapi spec has a format that allows for "extra" parameters to be added there with unknown names.  Types can be specified for those.  The explorer is less functional for this use case although the examples in the openapi spec are a bit obscure to me.  It is certainly possible that I am stating things wrongly.

See usage of "additionalProperties" in https://spec.openapis.org/oas/latest.html#parameter-object-examples

**Comment by adam-homeboost** [comment]
Minimal schema with examples for multipart/form-data POST and GET use cases:

```json
{
  "openapi": "3.1.0",
  "info": {
    "title": "Multipart upload schema",
    "version": "1"
  },
  "paths": {
    "/api/images": {
      "post": {
        "summary": "Upload Image",
        "parameters": [],
        "requestBody": {
          "content": {
            "multipart/form-data": {
              "schema": {
                "properties": {
                  "photo": {
                    "type": "string",
                    "format": "binary",
                    "title": "Photo"
                  }
                },
                "additionalProperties": true,
                "type": "object",
                "required": [
                  "photo"
                ]
              }
            }
          },
          "required": true
        }
      }
    },
    "/api/get-example": {
      "get": {
        "summary": "GET example",
        "parameters": [
          {
            "in": "query",
            "name": "freeForm",
            "schema": {
              "type": "object",
              "additionalProperties": true
            },
            "style": "form"
          }
        ]
      }
    }
  }
}
```

**Comment by wparad** [comment]
You got it mostly right. That's correct with the GET, the spec doesn't allow "unknown" query parameters, but it does allow a query parameter object. As you defined it this would be your expected input:

`https://path.server.com?freeForm=property1,value1,property2,value2`

Here the object freeForm is expected to look like:
```json
{
  property1: value1,
  property2: value2
}
```

I'm guessing though, didn't want the `freeForm` text in the query, and wanted this:
`https://path.server.com?property1=value1&property2=value2`

Note, this is not allowed in OAS 3.X. Before we add explicitly handling for this functionality, I want to confirm that is your expectation.

**Comment by adam-homeboost** [comment]
I'm a bit on the fence for the GET use case since I am finding the spec examples a little confusing.  The way you described it with explode makes a lot more sense to me and more closely matches what I would have expected.

However, I care far more about the multipart/form-data use case as that is the one I really need the most.  On the GET case I am happy to defer to those more expert than me and I can live without it.

**Comment by wparad** [comment]
Let's assign this issue for handling the form-data POST body and ignore the GET for now. If it comes up for someone in the future, we can revisit what the expected interpretation is. the form-data POST body is an obvious bug, and the user should be able to enter both the key and the value any number of times.

**Comment by wparad** [comment]
This has been fixed in `1.0.577`.
